startup.py

```diff
@@ -64,20 +64,65 @@ def scan_software(self, versions=None, display_name=None, icon=None):
 
     def prepare_launch(self, exec_path, args, file_to_open=None):
         """
-        Prepares the given software for launch.
+        Prepares an environment to launch Maya in that will automatically
+        load Toolkit and the tk-maya engine when Maya starts.
 
-        :param str exec_path: Path to DCC executable to launch.
+        :param str exec_path: Path to Maya executable to launch.
         :param str args: Command line arguments as strings.
         :param str file_to_open: (optional) Full path name of a file to open on launch.
         :returns: :class:`LaunchInformation` instance
         """
         required_env = {}
+
+        # Run the engine's userSetup.py file when Maya starts up
+        # by appending it to the env PYTHONPATH.
         startup_path = os.path.join(self.disk_location, "startup")
         sgtk.util.append_path_to_env_var("PYTHONPATH", startup_path)
         required_env["PYTHONPATH"] = os.environ["PYTHONPATH"]
-        required_env["SGTK_ENGINE"] = self.engine_name
-        required_env["SGTK_CONTEXT"] = sgtk.context.serialize(self.context)
+
+        # Check the engine settings to see whether any plugins have been
+        # specified to load.
+        load_plugins = self.get_setting("launch_builtin_plugin") or None
+        if load_plugins:
+            # Parse the specified comma-separated list of plugins
+            find_plugins = [p.strip() for p in load_plugins.split(",") if p.strip()]
+            for find_plugin in find_plugins:
+                load_plugin = os.path.join(
+                    self.disk_location, "plugins", find_plugin
+                )
+                if os.path.exists(load_plugin):
+                    # If the plugin path exists, add it to the list of MAYA_MODULE_PATHS
+                    # so Maya can find it and to the list of SGTK_LOAD_MAYA_PLUGINS so
+                    # the startup's userSetup.py file knows what plugins to load.
+                    self.logger.info("Loading builtin plugin '%s'" % load_plugin)
+                    sgtk.util.append_path_to_env_var("MAYA_MODULE_PATH", load_plugin)
+                    sgtk.util.append_path_to_env_var("SGTK_LOAD_MAYA_PLUGINS", load_plugin)
+                else:
+                    # Report the missing plugin directory
+                    self.logger.debug("Resolved plugin path '%s' does not exist!" %
+                        load_plugin
+                    )
+
+            # Add the MAYA_MODULE_PATH and SGTK_LOAD_MAYA_PLUGINS to the launch
+            # environment.
+            required_env["MAYA_MODULE_PATH"] = os.environ["MAYA_MODULE_PATH"]
```

Comment:
any reason we both add them to the environment and pass them back to the launch app? is it not enough just to pass them back tot he launch app?